### ISSUE TITLE
False positives on `prefer_declaring_const_constructors` when class extends a class that cannot be const.

### ISSUE THREAD

**Issue by arthurbcd** [solid-software/solid_lints#136]
The example below shows the false-positive:

```dart
class MyNotifier extends ChangeNotifier { // <- ChangeNotifier is a no-const class

// LINT (prefer_declaring_const_constructors)
MyNotifier(this.dependency);

final SomeDependency dependency;
}
```

It would be awesome if the linter knows that the extended class is not const.

If it's not possible, then maybe add an option to ignore if the class extends another. (and set it as default)
